### ISSUE TITLE
Lint to find std imports that could be imported as core or alloc

### ISSUE THREAD

**Issue by i509VCB** [rust-lang/rust-clippy#8917]
### What it does

If an import is a type exposed in `core` or `alloc`, the lint should warn that a the type could be imported using `core` or `alloc` instead.

### Lint Name

prefer_core_alloc_imports

### Category

pedantic

### Advantage

- For crates which support no_std environments, this makes finding accidental std import additions easier that could break the crate when compiling for a no_std environment.

### Drawbacks

- Crates which reexport core, alloc or std might behave differently.

### Example

```rust
use std::mem::size_of;
use std::vec::Vec;
```

Could be written as:

```rust
use core::mem::size_of;
use alloc::vec::Vec;
```

**Comment by Serial-ATA** [comment]
Duplicate of #7724

**Comment by giraffate** [comment]
Thanks for suggesting new lint!

I'm closing this because of https://github.com/rust-lang/rust-clippy/issues/8917#issuecomment-1141534554.
